Emby.Server.Implementations/ScheduledTasks/PluginUpdateTask.cs

```diff
@@ -0,0 +1,142 @@
+using MediaBrowser.Common;
+using MediaBrowser.Common.Updates;
+using MediaBrowser.Model.Net;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Common.Progress;
+using MediaBrowser.Model.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Emby.Server.Implementations.ScheduledTasks
+{
+    /// <summary>
+    /// Plugin Update Task
+    /// </summary>
+    public class PluginUpdateTask : IScheduledTask, IConfigurableScheduledTask
+    {
+        /// <summary>
+        /// The _logger
+        /// </summary>
+        private readonly ILogger _logger;
+
+        private readonly IInstallationManager _installationManager;
+
+        private readonly IApplicationHost _appHost;
+
+        public PluginUpdateTask(ILogger logger, IInstallationManager installationManager, IApplicationHost appHost)
+        {
+            _logger = logger;
```

Comment:
I think some of last PRs changed stuff to accept `ILoggerFactory` and create `logger` instance themselves. You probably have to follow that. Though I'm not 100% sure if that change hit the tasks...

Reply:
I only matched the other tasks so I am not sure this needs to change until all the other tasks are updated.